### DISCUSSION TITLE
Time dependence from csv file

### DISCUSSION THREAD

**Discussion by heinono1** [idaholab/moose#22184] [Q&A General] [answered]
I have a vector field (Hx,Hy,Hz) read in as auxiliary variables using the UO PropertyReadFile, then passed to Function node_Hx, node_Hy, node_Hz using PieceWiseConstantFromCSV, and then seed the (Hx,Hy,Hz) field from that function. What I then need to do is to multiply the field (Hx,Hy,Hz) with a time dependence that cannot be parameterized but is read into a PiecewiseMultilinearfunction that is used to define an auxiliary variable "coil_current". What I'd like to do is then to use an AuxKernel that multiplies "coil_current" (which is the value of a current given by times t from the PiecewiseMultilinear function) with the field (Hx,Hy,Hz) to get a field time-dependent field (Hext_x_time, Hext_y_time,Hext_z_time). I am not sure about the syntax in the AuxKernel. Right now I have
[AuxKernels]
[./Hext_x_time]
   type = FunctionAux
   variable = Hext_x
   function = 'coil_current*node_Hx'
[../]
and similarly for the y and z components. But this is not working (garbled syntax, I think). What is the correct syntax for multiplying together a time-dependent variable (coil_current) with a function (node_Hx)?

Thanks,
Olle

**Comment by heinono1** [comment, marked as answer]
It seems I can use CompositeFunction to multiply the two functions together to generate a new function that can be read in as a FunctionAux. At least running with --check-input throws no error.....

**Reply by GiudGiud** [reply]
Hello

That is one option. The other is to use the ParsedAux, that should let you multiply two functions together. 

Guillaume

**Comment by heinono1** [comment]
Thanks, Guillaume.

On Sat, Sep 24, 2022 at 8:43 PM Guillaume Giudicelli <
***@***.***> wrote:

> Hello
>
> That is one option. The other is to use the ParsedAux, that should let you
> multiply two functions together.
>
> Guillaume
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/discussions/22184#discussioncomment-3725611>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEKZEFY4U45L6CDDYMCZFITV76U4NANCNFSM6AAAAAAQUKJYBA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Olle Heinonen
***@***.***
